manifests.go

```diff
@@ -80,7 +81,17 @@ var mappings = make(map[string]UnmarshalFunc, 0)
 
 // UnmarshalManifest looks up manifest unmarshall functions based on
 // MediaType
-func UnmarshalManifest(mediatype string, p []byte) (Manifest, Descriptor, error) {
+func UnmarshalManifest(ctHeader string, p []byte) (Manifest, Descriptor, error) {
+	// Need to look up by the actual content type, not the raw contents of
+	// the header. Strip semicolons and anything following them.
+	var mediatype string
+	semicolonIndex := strings.Index(ctHeader, ";")
```

Comment:
Using the mime package would be better. Opened https://github.com/docker/distribution/pull/1376 for this.

Do you think it's worth revendoring distribution for Docker 1.10-rc2 once #1376 is merged? There could be parsing differences in some cases, such as `Content-Type: application/json   ;    charset=utf-8`. But I'm not sure this will ever be an issue in practice.